### PR TITLE
Add heroku deployment guide

### PR DESCRIPTION
**Pull request by zigomir**
I see there are many other guides for heroku proposed:

- https://github.com/vuejs/vue-cli/pull/3684
- https://github.com/vuejs/vue-cli/pull/3607
- https://github.com/vuejs/vue-cli/pull/3445
- https://github.com/vuejs/vue-cli/pull/3285
- https://github.com/vuejs/vue-cli/pull/2998
- https://github.com/vuejs/vue-cli/pull/3365

But all of them include some sort of node server which isn't really required with `heroku-buildpack-static` buildpack.

I'd argue that, this is the easiest and most straightforward way to deploy vue-cli app to heroku ;)

### CODE REVIEW

**Comment by aislanmaia** [comment]
So, how is the "easiest way"? I mean, how to deploy to heroku using vue-cli?

**Comment by zigomir** [comment]
@aislanmaia see insturcitons in https://github.com/vuejs/vue-cli/pull/3777/files

I just deployed [a fresh vue-cli app to heroku](https://pacific-lake-41226.herokuapp.com), without needing to add any other packages like other pull requests do – that's why I claim this is the easiest.

It also handles SPA correctly and serves `index.html` for all routes.

@sodatea whom should I bother to review this PR?

**Comment by aislanmaia** [comment]
Oh man.. many thanks for the instructions. I was right now trying to deploy with vue-cli but got error into the heroku app without the `serve` package, because heroku can't found vue-cli-service command. 
I found that I was missing to add `heroku-buildpack`.

**Comment by aislanmaia** [comment]
@zigomir what you put in the start script into `package.json`
Heroku logs error missing that part. 

Here is my scripts:
```
"scripts": {
    "serve": "vue-cli-service serve",
    "build": "vue-cli-service build",
    "lint": "vue-cli-service lint",
    "heroku-postbuild": "npm run build"
},
```

**Comment by zigomir** [comment]
@aislanmaia I think that adding `heroku buildpacks:add heroku/nodejs` will pick up `build` automatically. But would need to check.

**Comment by aislanmaia** [comment]
I have tested it out. It works with the buildpacks. But I have ditched Heroku because I'm using Cloudinary service to my images processing, and whatever you want, Heroku forces you to install addons. I'm using now firebase hosting that works straight forward with the libs my static frontend is using.

**Comment by NataliaTepluhina** [comment]
@zigomir thank you for creating this PR! However, it missed a step for Heroku CLI installation. Would you mind to add it to the guide?

**Comment by zigomir** [comment]
@NataliaTepluhina done. I just added a link as heroku has different installers for each platform.

File: docs/guide/deployment.md [hunk 1]
```diff
@@ -261,7 +261,35 @@ npm install -g now
...
+git add static.json
+git commit -m "add static configuration"
+```
+
+4. Deploy to Heroku
+```bash
+heroku login
+heroku create
+heroku buildpacks:add heroku/nodejs
+heroku buildpacks:add https://github.com/hone/heroku-buildpack-static
```
Comment (NataliaTepluhina) [inline docs/guide/deployment.md]:
```suggestion
heroku buildpacks:add https://github.com/heroku/heroku-buildpack-static
```
Previous URL redirects to Heroku organisation repo now.

**Comment by NataliaTepluhina** [comment]
@zigomir tested it and it looks like definitely the easiest way to deploy to Heroku. Left one minor suggestion, as soon as it's resolved I will merge.

File: docs/guide/deployment.md (hunk 1 above)
Comment (zigomir) [reply docs/guide/deployment.md]: Nice, thanks!

**Comment by zigomir** [comment]
@NataliaTepluhina thanks, done!

File: docs/guide/deployment.md [hunk 2]
```diff
@@ -261,7 +261,35 @@ npm install -g now
 
 ### Heroku
 
-> TODO | Open to contribution.
+1. [install HerokU CLI](https://devcenter.heroku.com/articles/heroku-cli)
```
Comment (NataliaTepluhina) [inline docs/guide/deployment.md]:
```suggestion
1. [Install Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli)
```

Comment (NataliaTepluhina) [reply docs/guide/deployment.md]: Sorry, one more suggestion (I would commit it myself but don't have access :) )

**Comment by neillsom** [comment]
When deploying to heroku I keep running into the error "`npm ERR! missing script: start`"

I was able to get the app running properly by adding a start script to package.json

```
	"scripts": {
		"serve": "vue-cli-service serve",
		"start": "vue-cli-service build",
		"build": "vue-cli-service build",
		"lint": "vue-cli-service lint"
	},
```

**Comment by zigomir** [comment]
@neillsom are you using both buildpacks mentioned in this PR?

```
heroku buildpacks:add heroku/nodejs
heroku buildpacks:add https://github.com/heroku/heroku-buildpack-static
```

**Comment by neillsom** [comment]
> @neillsom are you using both buildpacks mentioned in this PR?
> 
> ```
> heroku buildpacks:add heroku/nodejs
> heroku buildpacks:add https://github.com/heroku/heroku-buildpack-static
> ```

My mistake! I had github auto-deploying, so the heroku commands I was running from the command line were for the wrong remote 🤦 Thanks for helping me track it down

**Pull request merged**